modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NancyFXServerCodegen.java

```diff
@@ -46,6 +46,7 @@
     private static final String IMMUTABLE_OPTION = "immutable";
     private static final String USE_BASE_PATH = "writeModulePath";
     private static final String PACKAGE_CONTEXT = "packageContext";
+    private static final String ASYNC = "async";
```

Comment:
@craffael I would suggest using "supportsAsync" instead as the "supportsAsync" is used in C# client as well.

Reply:
In the C# client the meaning of `supportsAsync` is slightly different:
- It cannot be configured by the user through `--additional-properties` but is instead determined from e.g. the version of the .NET framework.
- If `supportsAsync=true` additional Code is generated if it is supported by the underlying .NET framework version. This makes sense because the C# client is a client, i.e. implementing every endpoint twice (once synchronous, once asynchronous) gives the user more choice. They can decide whether they want to call the sync or async method.

The last point is especially important: Following @jimschubert 's argument (https://github.com/swagger-api/swagger-codegen/issues/6734#issuecomment-337782969) I suggest a different approach for nancyfx because it generates not a client, but a server stub: The user should choose whether he wants to generate an async or a sync server stub (because one cannot implement an endpoint async *and* sync). Therefore its not about whether the environment (.NET Framework) *supports* async but whether the user *wants* async. Because of this I suggest to keep the name of this `--additional-property` as it is.

Reply:
@craffael what it sounds like @wing328 is getting at is that 'async' on its own may not be clear from CLI about what's actually being made async.

I agree supportsAsync would be confusing because this would give different semantics between client and server generators.

Could we do something like 'asyncServer' or 'asyncEndpoints'? This would be more maintainable in the long run. For example, if we were  (hypothetically) to add EF Code First Data layer and some data providers don't support async DB calls, we may want asynchronous endpoints, with synchronous DB calls. In such a case, just 'async' wouldn't work so we would have to introduce a breaking change to rename it.

Reply:
@jimschubert Okay I'm fine with `asyncServer` as well as `asyncEndpoints` but I prefer the `asyncServer` slightly. I will make the corresponding change in a moment.